syntax = "proto3";

package testsgen;

message Progress {
    string message = 1;
    double percent = 2;
    bool completed = 3;
}

message SourceCode {
    string filePath = 1;
    string code = 2;
}

message SourceInfo {
    string filePath = 1;
    uint32 line = 2;
}

enum ValidationType {
    UNSUPPORTED = 0;
    INT8_T = 1;
    INT16_T = 2;
    INT32_T = 3;
    INT64_T = 4;
    UINT8_T = 5;
    UINT16_T = 6;
    UINT32_T = 7;
    UINT64_T = 8;
    BOOL = 9;
    CHAR = 10;
    FLOAT = 11;
    STRING = 12;
}

message PredicateInfo {
    ValidationType type = 1;
    string predicate = 2;
    string returnValue = 3;
}
